lib/beaker-pe/install/ca_utils.rb

```diff
@@ -0,0 +1,188 @@
+#Much of this is taken from PuppetSpec:SSL
+require "openssl"
+
+module Beaker
+  module DSL
+    module InstallUtils
+      module CAUtils
+        PRIVATE_KEY_LENGTH = 2048
+        FIVE_YEARS = 5 * 365 * 24 * 60 * 60
+        CA_EXTENSIONS = [
+        ["basicConstraints", "CA:TRUE", true],
+        ["keyUsage", "keyCertSign, cRLSign", true],
+        ["subjectKeyIdentifier", "hash", false],
+        ["authorityKeyIdentifier", "keyid:always", false]
+        ]
+        NODE_EXTENSIONS = [
+        ["keyUsage", "digitalSignature", true],
+        ["subjectKeyIdentifier", "hash", false]
+        ]
+        DEFAULT_SIGNING_DIGEST = OpenSSL::Digest::SHA256.new
+        DEFAULT_REVOCATION_REASON = OpenSSL::OCSP::REVOKED_STATUS_KEYCOMPROMISE
+        ROOT_CA_NAME = "/CN=root-ca"
+        INT_CA_NAME = "/CN=intermediate-ca"
+        EXPLANATORY_TEXT = <<-EOT
+# Root Issuer: #{ROOT_CA_NAME}
+# Intermediate Issuer: #{INT_CA_NAME}
+EOT
+
+        # Generate CA bundle with root and intermediate certs, as well as CRL chain and private key for
+        # the intermediate CA, pushed to the soon-to-be master for use during PE install with pe_install::signing_ca
+        #
+        # @param [Host] host  The host to create CA bundle files on.  Defaults to global 'master' object.
+        # @param [String] targetdir  Location to save files on host, to be referenced in pe.conf for install.
+        def generate_ca_bundle_on(host = master, targetdir = '/tmp/ca_bundle')
```

Comment:
Maybe this should return a hash of the various certs/crls and the files they were written to? I like that you're writing the certs directly to the host. 👍

Reply:
What I'm thinking:
```
      {
        :root_cert => '/tmp/ca_bundle/root_cert',
        :int_cert => '/tmp/ca_bundle/int_cert',
        :int_ca_bundle => '/tmp/ca_bundle/int_ca_bundle',
        :int_key  => '/tmp/ca_bundle/int_key',
        :int_crl_chain => '/tmp/ca_bundle/int_crl_chain',
      }
```

Reply:
Good idea.  Were you thinking just filename => path/on/host ?  Or do you mean the SHA1 hash of each file after they are created?

Reply:
Good idea, I'll add that.